### PR TITLE
Add table aws_cloudfront_cache_policy. Closes #422

### LINKED ISSUE #422: Add table aws_cloudfront_cache_policy
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_cloudfront_cache_policy []

PRETEST: tests/aws_cloudfront_cache_policy

TEST: tests/aws_cloudfront_cache_policy
Running terraform
aws_cloudfront_cache_policy.named_test_resource: Creating...
aws_cloudfront_cache_policy.named_test_resource: Creation complete after 3s [id=c490c522-b25f-46f0-839d-960a855e6cff]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

etag = "E23ZP02F085DFQ"
resource_aka = "arn:aws:cloudfront::013122550996:cachepolicy/c490c522-b25f-46f0-839d-960a855e6cff"
resource_id = "c490c522-b25f-46f0-839d-960a855e6cff"
resource_name = "turbottest51809"

Running SQL query: test-get-query.sql
[
  {
    "comment": "turbottest51809",
    "etag": "E23ZP02F085DFQ",
    "id": "c490c522-b25f-46f0-839d-960a855e6cff",
    "name": "turbottest51809"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "comment": "turbottest51809",
    "etag": "E23ZP02F085DFQ",
    "id": "c490c522-b25f-46f0-839d-960a855e6cff",
    "name": "turbottest51809"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:cloudfront::013122550996:cachepolicy/c490c522-b25f-46f0-839d-960a855e6cff"
    ],
    "region": "global",
    "title": "turbottest51809"
  }
]
✔ PASSED

POSTTEST: tests/aws_cloudfront_cache_policy

TEARDOWN: tests/aws_cloudfront_cache_policy

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  id,
  name,
  comment,
  min_ttl,
  etag,
  last_modified_time
from
  aws_cloudfront_cache_policy;
+--------------------------------------+------------------------------------------------+-----------------------------------------------+---------+----------------+---------------------+
| id                                   | name                                           | comment                                       | min_ttl | etag          | last_modified_time  |
+--------------------------------------+------------------------------------------------+-----------------------------------------------+---------+----------------+---------------------+
| b2884449-e4de-46a7-ac36-70bc7f1ddd6d | Managed-CachingOptimizedForUncompressedObjects | Default policy when compression is disabled   | 1       | E23ZP02F085DFQ | 1970-01-01 00:00:00 |
| 4135ea2d-6df8-44a3-9df3-4b5a84be39ad | Managed-CachingDisabled                        | Policy with caching disabled                  | 0       | E23ZP02F085DFQ | 1970-01-01 00:00:00 |
| 08627262-05a9-4f76-9ded-b50ca2e3a84f | Managed-Elemental-MediaPackage                 | Policy for Elemental MediaPackage Origin      | 0       | E23ZP02F085DFQ | 1970-01-01 00:00:00 |
| 658327ea-f89d-4fab-a63d-7e88639e58f6 | Managed-CachingOptimized                       | Default policy when CF compression is enabled | 1       | E23ZP02F085DFQ | 1970-01-01 00:00:00 |
+--------------------------------------+------------------------------------------------+-----------------------------------------------+---------+----------------+---------------------+
> select
  id,
  name,
  parameters_in_cache_key_and_forwarded_to_origin ->> 'EnableAcceptEncodingBrotli' as enable_brotli
from
  aws_cloudfront_cache_policy
where
  parameters_in_cache_key_and_forwarded_to_origin ->> 'EnableAcceptEncodingBrotli' <> 'true';
+--------------------------------------+------------------------------------------------+---------------+
| id                                   | name                                           | enable_brotli |
+--------------------------------------+------------------------------------------------+---------------+
| b2884449-e4de-46a7-ac36-70bc7f1ddd6d | Managed-CachingOptimizedForUncompressedObjects | false         |
| 4135ea2d-6df8-44a3-9df3-4b5a84be39ad | Managed-CachingDisabled                        | false         |
| 08627262-05a9-4f76-9ded-b50ca2e3a84f | Managed-Elemental-MediaPackage                 | false         |
+--------------------------------------+------------------------------------------------+---------------+
> select
  id,
  name,
  parameters_in_cache_key_and_forwarded_to_origin ->> 'EnableAcceptEncodingGzip' as enable_gzip
from
  aws_cloudfront_cache_policy
where
  parameters_in_cache_key_and_forwarded_to_origin ->> 'EnableAcceptEncodingGzip' <> 'true';
+--------------------------------------+------------------------------------------------+-------------+
| id                                   | name                                           | enable_gzip |
+--------------------------------------+------------------------------------------------+-------------+
| b2884449-e4de-46a7-ac36-70bc7f1ddd6d | Managed-CachingOptimizedForUncompressedObjects | false       |
| 4135ea2d-6df8-44a3-9df3-4b5a84be39ad | Managed-CachingDisabled                        | false       |
+--------------------------------------+------------------------------------------------+-------------+

```
</details>

### CODE REVIEW

File: aws-test/tests/aws_cloudfront_cache_policy/variables.tf [hunk 1]
```diff
@@ -0,0 +1,91 @@
+variable "resource_name" {
+  type        = string
+  default     = "turbot-test-20200125-create-update"
+  description = "Name of the resource used throughout the test."
+}
+
+variable "aws_profile" {
+  type        = string
+  default     = "integration-tests"
```
Comment (khushboo9024) [inline aws-test/tests/aws_cloudfront_cache_policy/variables.tf:9]:
```suggestion
  default     = "default"
```

File: aws/table_aws_cloudfront_cache_policy.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+		Description: "AWS CloudFront Cache Policy",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.SingleColumn("id"),
+			ShouldIgnoreError: isNotFoundError([]string{"NoSuchCachePolicy"}),
+			Hydrate:           getCloudFrontCachePolicy,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listCloudFrontCachePolicies,
+		},
+		Columns: awsRegionalColumns([]*plugin.Column{
```
Comment (khushboo9024) [inline aws/table_aws_cloudfront_cache_policy.go:27]: This is not a regional resource, I don’t think we need to use `awsRegionalColumns`, please check .

File: aws/table_aws_cloudfront_cache_policy.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+				Transform:   transform.FromField("CachePolicy.CachePolicyConfig.Comment"),
+			},
+			{
+				Name:        "default_ttl",
+				Description: "The default amount of time, in seconds, that you want objects to stay in the CloudFront cache before CloudFront sends another request to the origin to see if the object has been updated.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromField("CachePolicy.CachePolicyConfig.DefaultTTL"),
+			},
+			{
+				Name:        "e_tag",
```
Comment (khushboo9024) [inline aws/table_aws_cloudfront_cache_policy.go:53]:
```suggestion
				Name:        "etag",
```

File: aws/table_aws_cloudfront_cache_policy.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+				Transform:   transform.FromField("CachePolicy.CachePolicyConfig.MinTTL"),
+			},
+			{
+				Name:        "last_modified_time",
+				Description: "The date and time when the cache policy was last modified.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("CachePolicy.LastModifiedTime"),
+			},
+			{
+				Name:        "parameters_in_cache_key_and_forwarded_to_origin",
```
Comment (khushboo9024) [inline aws/table_aws_cloudfront_cache_policy.go:77]: we are missing `type` column , please check.
```
	// Type is a required field
	Type *string `type:"string" required:"true" enum:"CachePolicyType"`
	// contains filtered or unexported fields
````

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**